File: 2021/2/main.jl
function p1()
    h, v = 0, 0
    for line in readlines("input.txt")
        op, x = split(line)
        x = parse(Int, x)
        if op == "forward"
            h += x
        elseif op == "up"
            v -= x
        else
            v += x
        end
    end
    println(h * v)
end

function p2()
    h, v, a = 0, 0, 0
    for line in readlines("input.txt")
        op, x = split(line)
        x = parse(Int, x)
        if op == "down"
            a += x
        elseif op == "up"
            a -= x
        else
            h += x
            v += x * a
        end
    end
    println(h * v)
end

p1()
p2()

File: 2021/2/main.py
s = [x.split() for x in open("input.txt").read().split("\n")]
x = y = 0
for d, i in s:
    i = int(i)
    if d == "down":
        y += i
    if d == "up":
        y -= i
    if d == "forward":
        x += i
print(x * y)

x = y = a = 0
for d, i in s:
    i = int(i)
    if d == "down":
        a -= i
    if d == "up":
        a += i
    if d == "forward":
        x += int(i)
        y += a * int(i)
print(x * y)